lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -191,65 +191,55 @@ sub get_data_from_parent_zone {
 
 
 sub _check_domain {
-    my ( $self, $dn, $type ) = @_;
+    my ( $self, $domain, $type ) = @_;
 
-    my $result = eval {
-        if ( !defined( $dn ) ) {
-            return ( $dn, { status => 'nok', message => encode_entities( "$type required" ) } );
-        }
+    if ( !defined( $domain ) ) {
+        return ( $domain, { status => 'nok', message => encode_entities( "$type required" ) } );
+    }
 
-        if ( $dn =~ m/[^[:ascii:]]+/ ) {
-            if ( Zonemaster::LDNS::has_idn() ) {
-                eval { $dn = Zonemaster::LDNS::to_idn( $dn ); };
-                if ( $@ ) {
-                    return (
-                        $dn,
-                        {
-                            status  => 'nok',
-                            message => encode_entities( "The domain name is not a valid IDNA string and cannot be converted to an A-label" )
-                        }
-                    );
-                }
-            }
-            else {
+    if ( $domain =~ m/[^[:ascii:]]+/ ) {
+        if ( Zonemaster::LDNS::has_idn() ) {
+            eval { $domain = Zonemaster::LDNS::to_idn( $domain ); };
+            if ( $@ ) {
                 return (
-                    $dn,
+                    $domain,
                     {
-                        status => 'nok',
-                        message =>
-                        encode_entities( "$type contains non-ascii characters and IDNA is not installed" )
+                        status  => 'nok',
+                        message => encode_entities( "The domain name is not a valid IDNA string and cannot be converted to an A-label" )
                     }
                 );
             }
         }
-
-        if( $dn !~ m/^[\-a-zA-Z0-9\.\_]+$/ ) {
+        else {
             return (
-                $dn,
+                $domain,
                 {
-                status  => 'nok',
-                message => encode_entities( "The domain name character(s) not supported")
+                    status  => 'nok',
+                    message => encode_entities( "$type contains non-ascii characters and IDNA is not installed" )
```

Comment:
"$type contains non-ascii label but support for conversion from U-label to A-label is not installed"